Drivers/imu/inc/icm42688.hpp

```diff
@@ -0,0 +1,115 @@
+        
+        /**
+        * Calibrates the gyroscope. Algorithm adapted from https://github.com/finani/ICM42688/tree/master/src
+         *
+        *  @return none 
+         */
+        void calibrate() override;
+        
+        /**
+        * Retrieve and process IMU accelerometer and gyroscope data into m/s and deg/s
+         *
+        * @param data_buffer -> array used to store and process raw data from IMU
+         * 
+        * @return IMUData_t structure consisting of accelerometer and gyroscope data in m/s and deg/s
+         */
+        IMUData_t getResult(uint8_t * data_buffer) override;
+
+    private:
+        /**
+        * Communicate with IMU via SPI to read raw data
+         * 
+        * @param sub_address -> memory address of starting byte to be retrieved
+        * @param num_bytes_to_retrieve -> number of bytes to be retrieved
+        * @param destination -> array to be populated with raw data
+         * 
+        * @return none
+         */
+        void readRegister(uint8_t sub_address, uint8_t num_bytes_to_retrieve, uint8_t * destination);
+        
+        /**
+        * Communicate with IMU via SPI to write data onto IMU
+         * 
+        * @param sub_address -> memory address of byte to be written on
+        * @param data_to_imu -> data to be written onto IMU
+         * 
+        * @return none
+         */
+        void writeRegister(uint8_t sub_address, uint8_t data_to_imu);
+
+        /**
+        * Resets the IMU's device configurations. Necessary to initilize IMU
+         * 
+        * @return none
+         */
+        void reset();
+
+        /**
+        * Sets the IMU's power management settings to low noise mode. Necessary to initilaize IMU
+         * 
+        * @return none
+         */
+        void setLowNoiseMode();
+
+        /**
+        * Sets the IMU register address bank to a value of 0 - 4
+         *
+        * @param bank -> Bank number (0 - 4)
+         * 
+        * @return none
+         */
+        void setBank(uint8_t bank);
+
+        /**
+        * Configures the full-scale range of the gyroscope on the IMU. Adapted from https://github.com/finani/ICM42688/tree/master/src
+         * 
+        * @param fssel -> full-scale selection for the gyro
+         * 
+        * @return none
+         */
+        void setGyroFS(uint8_t fssel);
+
+        //Variables used in gyro calibration
+        float gyro_scale = 0;       //gyro scale factor
+        uint8_t current_fssel = 0;  //current full-scale selection for the gyro
+        uint8_t gyroFS = 0;
+        float gyroBD[3] = {0, 0, 0};
+        float gyrB[3] = {0, 0, 0};
+        float gyr[3] = {0, 0, 0};
+        uint8_t gyro_buffer[GYRO_CALIBRATION_DATA_BUFFER_SIZE];
+        int16_t raw_meas_gyro[RAW_MEAS_BUFFER_SIZE];
+
+        //Used to hold raw IMU data
+        int16_t raw_meas[RAW_MEAS_BUFFER_SIZE];
+
+        SPI_HandleTypeDef * SPI_HANDLE;
+        GPIO_TypeDef * CS_GPIO_PORT;
+        uint16_t CS_PIN;
```

Comment:
It will be the best if you can change these member variables according to https://uwarg-docs.atlassian.net/wiki/spaces/ZP/pages/2268692486/Style+Standards . Having only these member variables all caps are weird. And also conventionally, ppl think all caps means macro.